docs/tasks/repo2docker-local.qmd

```diff
@@ -0,0 +1,64 @@
+
+## Common Usage
+
+One can simply run `repo2docker /path/to/image/assets`. For example if one has changed into the directory containing the `repo2docker` files (such as `environment.yml` and/or `Dockerfile`), the command would be:
+
+```shell
+repo2docker .
+```
+
+This works on Linux and Windows Subsystem for Linux (WSL). It will build the image, then launch jupyter server and display a localhost URL. Copy the URL and paste it into a local web browser.
+
+If you just want to build the image without also running the server,
+add the `--no-run` argument:
+
+```shell
+repo2docker --no-run .
+```
+
+## On Apple Silicon
+
+Apple's ARM-based CPUs (the "M" chips) are different from those run on the virtual machines in our clusters. macOS is capable of emulating x86_64/amd64, but it is necessary to optimize docker for this emulation, and to explicitly tell your local docker runtime that the images should be built on the `linux/amd64` platform.
+
+In Docker's settings:
+
+ - Under **General** > **Virtual Machine Options**, either enable both **Apple Virtualization framework** and **Use Rosetta for x86_64/amd64 emulation on Apple Silicon**, or enable **Docker VMM**.
+ - Under **Resources** it is also recommended to raise the memory limit to at least 4GB. 
+
+There are two methods for building `linux/amd64` images. The default uses `repo2docker`'s support for `docker-py`, while the second uses a `repo2docker` plugin that can invoke your local docker command-line interface.
+
+### docker-py (default)
+
+Run `jupyter-repo2docker` with the following arguments:
+
+```
+repo2docker \
+  --Repo2Docker.platform=linux/amd64 \
+  -e PLAYWRIGHT_BROWSERS_PATH=/srv/conda \
+  --user-id=1000 --user-name=jovyan \
+  --target-repo-dir=/home/jovyan/.cache \
+  .
+```
+
+where the final parameter is the path to the assets or `.` if they are in the current directory.
+
+The `--user-id` and `--user-name` options are for non-Dockerfile based builds. Images with Dockerfiles do not need those options.
+
+Note that you may see (possibly harmless) architecture mismatch warnings with this method.
+
+### `docker` CLI
+
+You can instruct `repo2docker` to use your machine's local `docker` executable directly rather than the default of `docker-py`. You will first need to install [repo2podman](https://github.com/manics/repo2podman), a plugin that lets you use any container runtime with a command-line user interface similar to that of `docker`. This is useful if you want to leverage [docker buildx](https://github.com/docker/buildx/) (for things like multi-stage builds) or if you want to use an alternative executable like `podman`. This also eliminates architecture mismatch warnings.
```

Comment:
@shaneknapp Was it failing on subprocess-related things? For now I can make a note about it, but it might be worth creating an issue in the repo2podman repo.

Reply:
here's the full stacktrace of the error:

```
(dh) ➜  a11y-user-image git:(main) repo2docker -e PLAYWRIGHT_BROWSERS_PATH=/srv/conda --user-id=1000 --user-name=jovyan \
  --target-repo-dir=/home/jovyan/.cache --engine podman --PodmanEngine.podman_executable=docker .
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
Picked Local content provider.
Using local repo ..
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
Building conda environment for python=3.11
Using CondaBuildPack builder
Traceback (most recent call last):
  File "/home/sknapp/miniconda3/envs/dh/bin/repo2docker", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/repo2docker/__main__.py", line 474, in main
    r2d.start()
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/repo2docker/app.py", line 893, in start
    self.build()
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/repo2docker/app.py", line 857, in build
    for l in picked_buildpack.build(
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/repo2docker/buildpacks/base.py", line 671, in build
    yield from client.build(**build_kwargs)
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/site-packages/repo2podman/podman.py", line 459, in build
    tarf.extractall(builddir)
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/tarfile.py", line 2294, in extractall
    self._extract_one(tarinfo, path, set_attrs=not tarinfo.isdir(),
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/tarfile.py", line 2361, in _extract_one
    self._handle_fatal_error(e)
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/tarfile.py", line 2357, in _extract_one
    self._extract_member(tarinfo, os.path.join(path, tarinfo.name),
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/tarfile.py", line 2440, in _extract_member
    self.makefile(tarinfo, targetpath)
  File "/home/sknapp/miniconda3/envs/dh/lib/python3.11/tarfile.py", line 2486, in makefile
    with bltn_open(targetpath, "wb") as target:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/tmp/tmpblydltvz/src/.git/objects/65/15cad7fcf550e9e34b9b762c687e021c085449'
```

same error if i run `repo2docker --engine podman --PodmanEngine.podman_executable=docker .`.

running `repo2docker .` in the same image directory works just fine.

Reply:
Okay, thanks. Odd that it is running into filesystem permission issues just extracting a tar file.